app/models/github_blob.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require 'safe_yaml/load'
+
+class GitHubBlob
+  # From jekyll/jekyll
+  # https://github.com/jekyll/jekyll/blob/74373baa550282a8630368e7b609ca9370f6d560/lib/jekyll/document.rb#L13
+  YAML_FRONT_MATTER_REGEXP = /\A(---\s*\n.*?\n?)^((---|\.\.\.)\s*$\n?)/m
+
+  attr_reader :data, :body
+
+  def initialize(github_repository, sha, **options)
+    repo = github_repository
+    @blob = repo.client.blob(repo.full_name, sha, options)
+    @data = nil
```

Comment:
This and the line below aren't really necessary since we defined the attr_reader.